### DISCUSSION TITLE
What are the free tier limits?

### DISCUSSION THREAD

**Discussion by richhenderson** [beginner-corp/begin-community#67] [Q&A] [answered]
Hi all!

I have enjoyed trying out begin so far but my hobby app quickly started hitting free tier limits, like 10 http functions. Now I am getting a build error:
```
⚬ Hydrate Hydrating dependencies in 6 paths
⚬ Hydrate Hydrating project root
× Error: Command failed: npm ci
npm WARN tar ENOSPC: no space left on device, write
npm ERR! cb() never called!

npm ERR! This is an error with npm itself. Please report this error at:
npm ERR!     <https://npm.community>
npm WARN tar ENOSPC: no space left on device, write
npm ERR! cb() never called!

npm ERR! This is an error with npm itself. Please report this error at:
npm ERR!     <https://npm.community>

    at ChildProcess.exithandler (child_process.js:308:12)
    at ChildProcess.emit (events.js:314:20)
    at ChildProcess.EventEmitter.emit (domain.js:483:12)
    at maybeClose (internal/child_process.js:1021:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)
```
Could this be that I have run out of free tier storage or have I broken something else like symlinks for modules that are taking up all the space? How much space is there?

I'm not sure what other limits I might run into. I am grateful for the free tier, but without knowing the limits, it is hard to know if an app is right for begin, before lots of time is spent on it.

Anyway, love begin, just very frustrated about the lack of transparency over free tier limits. Any help appreciated.

Thanks

**Comment by brianleroux** [comment, marked as answer]
Hey there sorry for the lack of clarity. Our limits are the AWS limits! Sometimes called quotas. Your application is composed of AWS Lambda, API Gateway and a DynamoDB table. You can see the Lambda limits here: https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-limits.html …the error you're getting is because your application code footprint is too large for the build (it has exceeded 500mb)

**Reply by richhenderson** [reply]
Thanks for the reply. I love what you have done with begin and I'm glad that I spent some time getting to know it. Although it isn't suitable for what I was try to do this time, I'll be back as soon as I have something that fits.
